### ISSUE TITLE
ImportError: cannot import name UnrewindableBodyError

### ISSUE THREAD

**Issue by fracture-point** [loomchild/volume-backup#20]
Hi there,

Love this tool!  I was using it without issue until yesterday; I now receive this error upon running which is difficult to debug:

```
[centos@oc]$ docker run -v owncloud_mysql:/volume --rm loomchild/volume-backup backup -| aws s3 cp - s3://{REMOVED-FOR-GITHUB}/owncloud_mysql-$(date +%F_%H-%M-%S).tar.bz2 --storage-class STANDARD_IA

Traceback (most recent call last):
  File "/usr/bin/aws", line 19, in <module>
    import awscli.clidriver
  File "/usr/lib/python2.7/site-packages/awscli/clidriver.py", line 17, in <module>
    import botocore.session
  File "/usr/lib/python2.7/site-packages/botocore/session.py", line 29, in <module>
    import botocore.configloader
  File "/usr/lib/python2.7/site-packages/botocore/configloader.py", line 19, in <module>
    from botocore.compat import six
  File "/usr/lib/python2.7/site-packages/botocore/compat.py", line 26, in <module>
    from urllib3 import exceptions
  File "/usr/lib/python2.7/site-packages/urllib3/__init__.py", line 10, in <module>
    from .connectionpool import (
  File "/usr/lib/python2.7/site-packages/urllib3/connectionpool.py", line 31, in <module>
    from .connection import (
  File "/usr/lib/python2.7/site-packages/urllib3/connection.py", line 45, in <module>
    from .util.ssl_ import (
  File "/usr/lib/python2.7/site-packages/urllib3/util/__init__.py", line 4, in <module>
    from .request import make_headers
  File "/usr/lib/python2.7/site-packages/urllib3/util/request.py", line 5, in <module>
    from ..exceptions import UnrewindableBodyError
ImportError: cannot import name UnrewindableBodyError
write /dev/stdout: broken pipe
```

As you can see, I'm backing up a mysql volume on a centos box.  The only things that have changed (though I'm not sure they'd impact the run) are:
1. I moved the default docker volumes directory to an external drive and sym-linked the directory.  This has not presented any other issues with docker or running containers.
2. I upgraded a ton of packages on the host os (listed below).  I only mention because of the python2.7 errors above.

```
Packages Altered:

    Updated     GeoIP-1.5.0-13.el7.x86_64                           ?
    Update            1.5.0-14.el7.x86_64                           @base
    Updated     audit-2.8.4-4.el7.x86_64                            ?
    Update            2.8.5-4.el7.x86_64                            @base
    Updated     audit-libs-2.8.4-4.el7.x86_64                       ?
    Update                 2.8.5-4.el7.x86_64                       @base
    Updated     audit-libs-python-2.8.4-4.el7.x86_64                ?
    Update                        2.8.5-4.el7.x86_64                @base
    Updated     bash-4.2.46-31.el7.x86_64                           ?
    Update           4.2.46-33.el7.x86_64                           @base
    Dep-Install bind-export-libs-32:9.11.4-9.P2.el7.x86_64          @base
    Updated     bind-libs-32:9.9.4-74.el7_6.2.x86_64                @updates
    Update                32:9.11.4-9.P2.el7.x86_64                 @base
    Updated     bind-libs-lite-32:9.9.4-74.el7_6.2.x86_64           ?
    Update                     32:9.11.4-9.P2.el7.x86_64            @base
    Updated     bind-license-32:9.9.4-74.el7_6.2.noarch             ?
    Update                   32:9.11.4-9.P2.el7.noarch              @base
    Updated     bind-utils-32:9.9.4-74.el7_6.2.x86_64               @updates
    Update                 32:9.11.4-9.P2.el7.x86_64                @base
    Updated     binutils-2.27-34.base.el7.x86_64                    ?
    Update               2.27-41.base.el7.x86_64                    @base
    Updated     centos-release-7-6.1810.2.el7.centos.x86_64         ?
    Update                     7-7.1908.0.el7.centos.x86_64         @base
    Updated     chrony-3.2-2.el7.x86_64                             ?
    Update             3.4-1.el7.x86_64                             @base
    Updated     cloud-init-18.2-1.el7.centos.2.x86_64               ?
    Update                 18.5-3.el7.centos.x86_64                 @base
    Updated     cloud-utils-growpart-0.29-2.el7_6.2.noarch          ?
    Update                           0.29-5.el7.noarch              @base
    Updated     container-selinux-2:2.107-1.el7_6.noarch            @extras
    Update                        2:2.107-3.el7.noarch              @extras
    Updated     coreutils-8.22-23.el7.x86_64                        ?
    Update                8.22-24.el7.x86_64                        @base
    Updated     cronie-1.4.11-20.el7_6.x86_64                       ?
    Update             1.4.11-23.el7.x86_64                         @base
    Updated     cronie-anacron-1.4.11-20.el7_6.x86_64               ?
    Update                     1.4.11-23.el7.x86_64                 @base
    Updated     cryptsetup-libs-2.0.3-3.el7.x86_64                  ?
    Update                      2.0.3-5.el7.x86_64                  @base
    Updated     curl-7.29.0-51.el7_6.3.x86_64                       ?
    Update           7.29.0-54.el7.x86_64                           @base
    Updated     device-mapper-7:1.02.149-10.el7_6.8.x86_64          ?
    Update                    7:1.02.158-2.el7.x86_64               @base
    Updated     device-mapper-libs-7:1.02.149-10.el7_6.8.x86_64     ?
    Update                         7:1.02.158-2.el7.x86_64          @base
    Updated     dhclient-12:4.2.5-68.el7.centos.1.x86_64            ?
    Update               12:4.2.5-77.el7.centos.x86_64              @base
    Updated     dhcp-common-12:4.2.5-68.el7.centos.1.x86_64         ?
    Update                  12:4.2.5-77.el7.centos.x86_64           @base
    Updated     dhcp-libs-12:4.2.5-68.el7.centos.1.x86_64           ?
    Update                12:4.2.5-77.el7.centos.x86_64             @base
    Updated     diffutils-3.3-4.el7.x86_64                          ?
    Update                3.3-5.el7.x86_64                          @base
    Updated     dmidecode-1:3.1-2.el7.x86_64                        ?
    Update                1:3.2-3.el7.x86_64                        @base
    Updated     dracut-033-554.el7.x86_64                           ?
    Update             033-564.el7.x86_64                           @base
    Updated     dracut-config-generic-033-554.el7.x86_64            ?
    Update                            033-564.el7.x86_64            @base
    Updated     dracut-config-rescue-033-554.el7.x86_64             ?
    Update                           033-564.el7.x86_64             @base
    Updated     dracut-network-033-554.el7.x86_64                   ?
    Update                     033-564.el7.x86_64                   @base
    Updated     e2fsprogs-1.42.9-13.el7.x86_64                      ?
    Update                1.42.9-16.el7.x86_64                      @base
    Updated     e2fsprogs-libs-1.42.9-13.el7.x86_64                 ?
    Update                     1.42.9-16.el7.x86_64                 @base
    Updated     elfutils-default-yama-scope-0.172-2.el7.noarch      ?
    Update                                  0.176-2.el7.noarch      @base
    Updated     elfutils-libelf-0.172-2.el7.x86_64                  ?
    Update                      0.176-2.el7.x86_64                  @base
    Updated     elfutils-libs-0.172-2.el7.x86_64                    ?
    Update                    0.176-2.el7.x86_64                    @base
    Updated     ethtool-2:4.8-9.el7.x86_64                          ?
    Update              2:4.8-10.el7.x86_64                         @base
    Updated     freetype-2.8-12.el7_6.1.x86_64                      ?
    Update               2.8-14.el7.x86_64                          @base
    Dep-Install geoipupdate-2.5.0-1.el7.x86_64                      @base
    Updated     glib2-2.56.1-4.el7_6.x86_64                         ?
    Update            2.56.1-5.el7.x86_64                           @base
    Updated     glibc-2.17-260.el7_6.6.x86_64                       ?
    Update            2.17-292.el7.x86_64                           @base
    Updated     glibc-common-2.17-260.el7_6.6.x86_64                ?
    Update                   2.17-292.el7.x86_64                    @base
    Updated     grub2-1:2.02-0.76.el7.centos.1.x86_64               ?
    Update            1:2.02-0.80.el7.centos.x86_64                 @base
    Updated     grub2-common-1:2.02-0.76.el7.centos.1.noarch        ?
    Update                   1:2.02-0.80.el7.centos.noarch          @base
    Updated     grub2-pc-1:2.02-0.76.el7.centos.1.x86_64            ?
    Update               1:2.02-0.80.el7.centos.x86_64              @base
    Updated     grub2-pc-modules-1:2.02-0.76.el7.centos.1.noarch    ?
    Update                       1:2.02-0.80.el7.centos.noarch      @base
    Updated     grub2-tools-1:2.02-0.76.el7.centos.1.x86_64         ?
    Update                  1:2.02-0.80.el7.centos.x86_64           @base
    Updated     grub2-tools-extra-1:2.02-0.76.el7.centos.1.x86_64   ?
    Update                        1:2.02-0.80.el7.centos.x86_64     @base
    Updated     grub2-tools-minimal-1:2.02-0.76.el7.centos.1.x86_64 ?
    Update                          1:2.02-0.80.el7.centos.x86_64   @base
    Updated     grubby-8.28-25.el7.x86_64                           ?
    Update             8.28-26.el7.x86_64                           @base
    Updated     gssproxy-0.7.0-21.el7.x86_64                        ?
    Update               0.7.0-26.el7.x86_64                        @base
    Updated     hwdata-0.252-9.1.el7.x86_64                         ?
    Update             0.252-9.3.el7.x86_64                         @base
    Updated     initscripts-9.49.46-1.el7.x86_64                    ?
    Update                  9.49.47-1.el7.x86_64                    @base
    Updated     iproute-4.11.0-14.el7_6.2.x86_64                    ?
    Update              4.11.0-25.el7.x86_64                        @base
    Updated     iptables-1.4.21-28.el7.x86_64                       ?
    Update               1.4.21-33.el7.x86_64                       @base
    Updated     irqbalance-3:1.0.7-11.el7.x86_64                    ?
    Update                 3:1.0.7-12.el7.x86_64                    @base
    Updated     iwl7265-firmware-22.0.7.0-69.el7.noarch             ?
    Update                       22.0.7.0-72.el7.noarch             @base
    Install     kernel-3.10.0-1062.1.1.el7.x86_64                   @updates
    Updated     kernel-tools-3.10.0-957.27.2.el7.x86_64             ?
    Update                   3.10.0-1062.1.1.el7.x86_64             @updates
    Updated     kernel-tools-libs-3.10.0-957.27.2.el7.x86_64        ?
    Update                        3.10.0-1062.1.1.el7.x86_64        @updates
    Updated     kexec-tools-2.0.15-21.el7_6.4.x86_64                ?
    Update                  2.0.15-33.el7.x86_64                    @base
    Updated     kmod-20-23.el7.x86_64                               ?
    Update           20-25.el7.x86_64                               @base
    Updated     kmod-libs-20-23.el7.x86_64                          ?
    Update                20-25.el7.x86_64                          @base
    Updated     kpartx-0.4.9-123.el7.x86_64                         ?
    Update             0.4.9-127.el7.x86_64                         @base
    Updated     krb5-libs-1.15.1-37.el7_6.x86_64                    @?base
    Update                1.15.1-37.el7_7.2.x86_64                  @updates
    Updated     libblkid-2.23.2-59.el7_6.1.x86_64                   ?
    Update               2.23.2-61.el7.x86_64                       @base
    Updated     libcap-2.22-9.el7.x86_64                            ?
    Update             2.22-10.el7.x86_64                           @base
    Updated     libcgroup-0.41-20.el7.x86_64                        ?
    Update                0.41-21.el7.x86_64                        @base
    Updated     libcom_err-1.42.9-13.el7.x86_64                     ?
    Update                 1.42.9-16.el7.x86_64                     @base
    Updated     libcurl-7.29.0-51.el7_6.3.x86_64                    ?
    Update              7.29.0-54.el7.x86_64                        @base
    Updated     libdb-5.3.21-24.el7.x86_64                          ?
    Update            5.3.21-25.el7.x86_64                          @base
    Updated     libdb-utils-5.3.21-24.el7.x86_64                    ?
    Update                  5.3.21-25.el7.x86_64                    @base
    Updated     libgcc-4.8.5-36.el7_6.2.x86_64                      ?
    Update             4.8.5-39.el7.x86_64                          @base
    Updated     libgomp-4.8.5-36.el7_6.2.x86_64                     ?
    Update              4.8.5-39.el7.x86_64                         @base
    Updated     libmount-2.23.2-59.el7_6.1.x86_64                   ?
    Update               2.23.2-61.el7.x86_64                       @base
    Updated     libndp-1.2-7.el7.x86_64                             ?
    Update             1.2-9.el7.x86_64                             @base
    Updated     libsmartcols-2.23.2-59.el7_6.1.x86_64               ?
    Update                   2.23.2-61.el7.x86_64                   @base
    Updated     libss-1.42.9-13.el7.x86_64                          ?
    Update            1.42.9-16.el7.x86_64                          @base
    Updated     libssh2-1.4.3-12.el7_6.3.x86_64                     ?
    Update              1.8.0-3.el7.x86_64                          @base
    Updated     libstdc++-4.8.5-36.el7_6.2.x86_64                   ?
    Update                4.8.5-39.el7.x86_64                       @base
    Updated     libteam-1.27-6.el7_6.1.x86_64                       ?
    Update              1.27-9.el7.x86_64                           @base
    Updated     libtirpc-0.2.4-0.15.el7.x86_64                      ?
    Update               0.2.4-0.16.el7.x86_64                      @base
    Updated     libuuid-2.23.2-59.el7_6.1.x86_64                    ?
    Update              2.23.2-61.el7.x86_64                        @base
    Dep-Install linux-firmware-20190429-72.gitddde598.el7.noarch    @base
    Updated     lshw-B.02.18-12.el7.x86_64                          ?
    Update           B.02.18-13.el7.x86_64                          @base
    Updated     lz4-1.7.5-2.el7.x86_64                              ?
    Update          1.7.5-3.el7.x86_64                              @base
    Updated     make-1:3.82-23.el7.x86_64                           ?
    Update           1:3.82-24.el7.x86_64                           @base
    Updated     mariadb-libs-1:5.5.60-1.el7_5.x86_64                ?
    Update                   1:5.5.64-1.el7.x86_64                  @base
    Updated     microcode_ctl-2:2.1-47.5.el7_6.x86_64               ?
    Update                    2:2.1-53.el7.x86_64                   @base
    Updated     net-tools-2.0-0.24.20131004git.el7.x86_64           ?
    Update                2.0-0.25.20131004git.el7.x86_64           @base
    Updated     nfs-utils-1:1.3.0-0.61.el7.x86_64                   ?
    Update                1:1.3.0-0.65.el7.x86_64                   @base
    Updated     nmap-ncat-2:6.40-16.el7.x86_64                      @base
    Update                2:6.40-19.el7.x86_64                      @base
    Updated     nspr-4.19.0-1.el7_5.x86_64                          ?
    Update           4.21.0-1.el7.x86_64                            @base
    Updated     nss-3.36.0-7.1.el7_6.x86_64                         ?
    Update          3.44.0-4.el7.x86_64                             @base
    Updated     nss-pem-1.0.3-5.el7_6.1.x86_64                      ?
    Update              1.0.3-7.el7.x86_64                          @base
    Updated     nss-softokn-3.36.0-5.el7_5.x86_64                   ?
    Update                  3.44.0-5.el7.x86_64                     @base
    Updated     nss-softokn-freebl-3.36.0-5.el7_5.x86_64            ?
    Update                         3.44.0-5.el7.x86_64              @base
    Updated     nss-sysinit-3.36.0-7.1.el7_6.x86_64                 ?
    Update                  3.44.0-4.el7.x86_64                     @base
    Updated     nss-tools-3.36.0-7.1.el7_6.x86_64                   ?
    Update                3.44.0-4.el7.x86_64                       @base
    Updated     nss-util-3.36.0-1.1.el7_6.x86_64                    ?
    Update               3.44.0-3.el7.x86_64                        @base
    Updated     numactl-libs-2.0.9-7.el7.x86_64                     ?
    Update                   2.0.12-3.el7.x86_64                    @base
    Updated     openssh-7.4p1-16.el7.x86_64                         ?
    Update              7.4p1-21.el7.x86_64                         @base
    Updated     openssh-clients-7.4p1-16.el7.x86_64                 ?
    Update                      7.4p1-21.el7.x86_64                 @base
    Updated     openssh-server-7.4p1-16.el7.x86_64                  ?
    Update                     7.4p1-21.el7.x86_64                  @base
    Updated     openssl-1:1.0.2k-16.el7_6.1.x86_64                  ?
    Update              1:1.0.2k-19.el7.x86_64                      @base
    Updated     openssl-libs-1:1.0.2k-16.el7_6.1.x86_64             ?
    Update                   1:1.0.2k-19.el7.x86_64                 @base
    Updated     parted-3.1-29.el7.x86_64                            ?
    Update             3.1-31.el7.x86_64                            @base
    Updated     passwd-0.79-4.el7.x86_64                            ?
    Update             0.79-5.el7.x86_64                            @base
    Updated     policycoreutils-2.5-29.el7_6.1.x86_64               ?
    Update                      2.5-33.el7.x86_64                   @base
    Updated     policycoreutils-python-2.5-29.el7_6.1.x86_64        ?
    Update                             2.5-33.el7.x86_64            @base
    Updated     polkit-0.112-18.el7_6.1.x86_64                      ?
    Update             0.112-22.el7_7.1.x86_64                      @updates
    Updated     procps-ng-3.3.10-23.el7.x86_64                      ?
    Update                3.3.10-26.el7.x86_64                      @base
    Updated     python-2.7.5-80.el7_6.x86_64                        ?
    Update             2.7.5-86.el7.x86_64                          @base
    Updated     python-chardet-2.2.1-1.el7_1.noarch                 ?
    Update                     2.2.1-3.el7.noarch                   @base
    Updated     python-jinja2-2.7.2-3.el7_6.noarch                  ?
    Update                    2.7.2-4.el7.noarch                    @base
    Updated     python-libs-2.7.5-80.el7_6.x86_64                   ?
    Update                  2.7.5-86.el7.x86_64                     @base
    Updated     python-linux-procfs-0.4.9-4.el7.noarch              ?
    Update                          0.4.11-4.el7.noarch             @base
    Updated     python-perf-3.10.0-957.27.2.el7.x86_64              ?
    Update                  3.10.0-1062.1.1.el7.x86_64              @updates
    Updated     python-requests-2.6.0-1.el7_1.noarch                ?
    Update                      2.6.0-5.el7.noarch                  @base
  Updated     python-urllib3-1.10.2-5.el7.noarch                  installed
  Update                     1.10.2-7.el7.noarch                  @?base
    Updated     qemu-guest-agent-10:2.12.0-2.el7.x86_64             ?
    Update                       10:2.12.0-3.el7.x86_64             @base
    Updated     quota-1:4.01-17.el7.x86_64                          ?
    Update            1:4.01-19.el7.x86_64                          @base
    Updated     quota-nls-1:4.01-17.el7.noarch                      ?
    Update                1:4.01-19.el7.noarch                      @base
    Updated     readline-6.2-10.el7.x86_64                          ?
    Update               6.2-11.el7.x86_64                          @base
    Updated     rpcbind-0.2.0-47.el7.x86_64                         ?
    Update              0.2.0-48.el7.x86_64                         @base
    Updated     rpm-4.11.3-35.el7.x86_64                            ?
    Update          4.11.3-40.el7.x86_64                            @base
    Updated     rpm-build-libs-4.11.3-35.el7.x86_64                 ?
    Update                     4.11.3-40.el7.x86_64                 @base
    Updated     rpm-libs-4.11.3-35.el7.x86_64                       ?
    Update               4.11.3-40.el7.x86_64                       @base
    Updated     rpm-python-4.11.3-35.el7.x86_64                     ?
    Update                 4.11.3-40.el7.x86_64                     @base
    Updated     rsyslog-8.24.0-34.el7.x86_64                        ?
    Update              8.24.0-41.el7_7.x86_64                      @updates
    Updated     selinux-policy-3.13.1-229.el7_6.15.noarch           ?
    Update                     3.13.1-252.el7.1.noarch              @updates
    Updated     selinux-policy-targeted-3.13.1-229.el7_6.15.noarch  ?
    Update                              3.13.1-252.el7.1.noarch     @updates
    Updated     sg3_utils-1.37-17.el7.x86_64                        ?
    Update                1.37-18.el7_7.1.x86_64                    @updates
    Updated     sg3_utils-libs-1.37-17.el7.x86_64                   ?
    Update                     1.37-18.el7_7.1.x86_64               @updates
    Updated     shadow-utils-2:4.1.5.1-25.el7_6.1.x86_64            ?
    Update                   2:4.6-5.el7.x86_64                     @base
    Updated     sudo-1.8.23-3.el7.x86_64                            ?
    Update           1.8.23-4.el7.x86_64                            @base
    Updated     systemd-219-62.el7_6.9.x86_64                       ?
    Update              219-67.el7_7.1.x86_64                       @updates
    Updated     systemd-libs-219-62.el7_6.9.x86_64                  ?
    Update                   219-67.el7_7.1.x86_64                  @updates
    Updated     systemd-sysv-219-62.el7_6.9.x86_64                  ?
    Update                   219-67.el7_7.1.x86_64                  @updates
    Updated     teamd-1.27-6.el7_6.1.x86_64                         ?
    Update            1.27-9.el7.x86_64                             @base
    Updated     tuned-2.10.0-6.el7_6.4.noarch                       ?
    Update            2.11.0-5.el7_7.1.noarch                       @updates
    Updated     util-linux-2.23.2-59.el7_6.1.x86_64                 ?
    Update                 2.23.2-61.el7.x86_64                     @base
    Updated     vim-minimal-2:7.4.160-6.el7_6.x86_64                ?
    Update                  2:7.4.629-6.el7.x86_64                  @base
    Updated     xfsprogs-4.5.0-19.el7_6.x86_64                      ?
    Update               4.5.0-20.el7.x86_64                        @base
    Updated     yum-3.4.3-161.el7.centos.noarch                     ?
    Update          3.4.3-163.el7.centos.noarch                     @base
    Updated     yum-plugin-fastestmirror-1.1.31-50.el7.noarch       ?
    Update                               1.1.31-52.el7.noarch       @base
    Updated     yum-utils-1.1.31-50.el7.noarch                      ?
    Update                1.1.31-52.el7.noarch                      @base
```

**Comment by loomchild** [comment]
Are you sure it's a problem with volume-backup? Can you try to do it without piping to aws CLI? Does it work in this case?

I don't use Python in volume-backup, it's written in bash, therefore I can't really see what is the relationship of my tool to the error - it looks more like your upgrade broke aws CLI (although I might be wrong).

**Comment by fracture-point** [comment]
Hey great catch!  You're right, upgrading python broke the AWS CLI.  I worked through 2 solutions, which ultimately fixed the issue.

1st one - fixed one library issue, and revealed another
https://ask.openstack.org/en/question/110437/importerror-cannot-import-name-unrewindablebodyerror/

Code I ran:
```
sudo pip uninstall requests
sudo pip uninstall urllib3
sudo yum remove python-urllib3
sudo yum remove python-requests
sudo yum install python-urllib3
sudo yum install python-requests
```


2nd one - fixed everything
https://stackoverflow.com/questions/55769854/postman-request-importerror-cannot-import-name-default-ciphers

Code I ran:
```
sudo yum remove PyYAML
sudo pip install awscli --force-reinstall --upgrade
```

Seems that the AWS installer like to manage dependancies via pip, so if any of them are already satisfied by yum packages, it blows up on upgrade.

Thanks again, and great tool!
